lib/spark_api/reso_faraday_middleware.rb

```diff
@@ -0,0 +1,30 @@
+
+module SparkApi
+
+  class ResoFaradayMiddleware < FaradayMiddleware
+
+    def on_complete(env)
+
+      body = decompress_body(env)
+
+      begin
+        body = MultiJson.decode(body)
+
+        if body["D"]
+          super(env)
+          return
+        end
+
+        env[:body] = body
+      rescue MultiJson::ParseError => e
```

Comment:
I'm a bit confused as to what this accomplishes, do we just swallow the error if we encounter `xml`?